vmm/src/vm.rs

```diff
@@ -576,10 +576,25 @@ impl Vm {
     }
 
     pub fn shutdown(&mut self) -> Result<()> {
-        let mut state = self.state.try_write().map_err(|_| Error::PoisonedState)?;
+        let current_state = self.get_state()?;
         let new_state = VmState::Shutdown;
 
-        state.valid_transition(new_state)?;
+        current_state.valid_transition(new_state)?;
+        // If the current state is paused that means most of the handles got killed by pthread_kill
+        // We need to unpark those threads
+        if current_state == VmState::Paused {
+            self.device_manager
+                .lock()
+                .unwrap()
+                .resume()
+                .map_err(Error::ResumeDevices)?;
+
+            self.cpu_manager
+                .lock()
+                .unwrap()
+                .resume()
+                .map_err(Error::ResumeCpus)?;
+        }
```

Comment:
I'd rather simply call into `self.resume()` when we're paused and are trying to shutdown the VMM.

Reply:
Thanks for reviewing. Addressed you comment.